<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>900</width>
    <height>827</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>841</width>
      <height>711</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <property name="tabsClosable">
     <bool>true</bool>
    </property>
    <property name="movable">
     <bool>true</bool>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>Tab 1</string>
     </attribute>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>620</x>
        <y>30</y>
        <width>202</width>
        <height>531</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Manage STLs currently in this world</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QListView" name="currentSTLs"/>
       </item>
       <item>
        <widget class="QPushButton" name="insertObject">
         <property name="text">
          <string>Insert additional STLs into this world</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="removeObject">
         <property name="text">
          <string>Remove selected STL</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>Recent Files</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QListView" name="recentFiles"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget_3">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>460</y>
        <width>601</width>
        <height>25</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QCheckBox" name="shrinkFilter">
         <property name="text">
          <string>Apply Shrink Filter</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_3">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>580</y>
        <width>601</width>
        <height>61</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_7">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QPushButton" name="cutObject">
           <property name="text">
            <string>Cutter</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="displayProperties">
           <property name="text">
            <string>Display properties</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="horizontalSpacer_7">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <spacer name="horizontalSpacer_8">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <widget class="QSpinBox" name="xCoordinate">
             <property name="maximum">
              <number>999</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label">
             <property name="text">
              <string>X</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="yCoordinate">
             <property name="maximum">
              <number>999</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Y</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="zCoordinate">
             <property name="maximum">
              <number>999</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Z</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QPushButton" name="changePosition">
           <property name="text">
            <string>Change Coordinates</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget_4">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>490</y>
        <width>601</width>
        <height>25</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <item>
        <widget class="QCheckBox" name="clipFilter">
         <property name="text">
          <string>Apply Clip Filter</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_4">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>20</y>
        <width>602</width>
        <height>434</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QVTKOpenGLWidget" name="qvtkWidget" native="true">
         <property name="minimumSize">
          <size>
           <width>600</width>
           <height>400</height>
          </size>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QPushButton" name="changeModelColor">
           <property name="text">
            <string>Change Model Color</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="changeBackgroundColor">
           <property name="text">
            <string>Change Background Color</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="renderCube">
           <property name="text">
            <string>Render Cube</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="renderPyramid">
           <property name="text">
            <string>Render Pyramid</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget_5">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>520</y>
        <width>601</width>
        <height>26</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_5">
       <item>
        <widget class="QCheckBox" name="outlineFilter">
         <property name="text">
          <string>Apply Outline Filter</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="changeOutlineColor">
         <property name="text">
          <string>Change Outline Color</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_5">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget_6">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>550</y>
        <width>601</width>
        <height>25</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_6">
       <item>
        <widget class="QCheckBox" name="edgeVisibilityFilter">
         <property name="text">
          <string>Apply Edge Visibility</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_6">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>900</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionFileOpen"/>
    <addaction name="actionFileSave"/>
    <addaction name="newWindowButton"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionFileOpen"/>
   <addaction name="actionFileSave"/>
   <addaction name="actionHelp"/>
   <addaction name="actionPrint"/>
  </widget>
  <action name="actionFileOpen">
   <property name="icon">
    <iconset resource="Icons/icons.qrc">
     <normaloff>:/Icons/fileopen.png</normaloff>:/Icons/fileopen.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Open</string>
   </property>
  </action>
  <action name="actionFileSave">
   <property name="icon">
    <iconset resource="Icons/icons.qrc">
     <normaloff>:/Icons/filesave.png</normaloff>:/Icons/filesave.png</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionHelp">
   <property name="icon">
    <iconset resource="Icons/icons.qrc">
     <normaloff>:/Icons/help.png</normaloff>:/Icons/help.png</iconset>
   </property>
   <property name="text">
    <string>Help</string>
   </property>
  </action>
  <action name="actionPrint">
   <property name="icon">
    <iconset resource="Icons/icons.qrc">
     <normaloff>:/Icons/print.png</normaloff>:/Icons/print.png</iconset>
   </property>
   <property name="text">
    <string>Print</string>
   </property>
  </action>
  <action name="newWindowButton">
   <property name="text">
    <string>New Window</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QVTKOpenGLWidget</class>
   <extends>QWidget</extends>
   <header>QVTKOpenGLWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="Icons/icons.qrc"/>
 </resources>
 <connections/>
</ui>
